### PR TITLE
[Kernel][Diffusion] Fuse Qwen-Image residual and paired AdaLN pointwise stages

### PR DESCRIPTION
**Pull request by 0z5a**
## Summary

Fuse Qwen-Image residual and AdaLN pointwise stages around the existing native FP32 LayerNorm reductions. The paired path combines the independent image/text pointwise launches while preserving each stream's original shape, epsilon, rounding boundaries, output storage, and MLP/TP-collective order. Scalar Triton launch metadata also supports dynamic-shape compilation.

Ordinary text-to-image inference can opt into the full pointwise path with `VLLM_OMNI_QWEN_ADALN_V2=1` and paired launches with `VLLM_OMNI_QWEN_ADALN_PAIR=1`. Indexed Edit, zero-condition, unsupported dtype/layout/device, and training cases retain their existing fallback. No concatenation, caller-input mutation, or persistent tensor-buffer cache is introduced.

## Validation

The submitted implementation is `d786da3`, including the five follow-up commits on this existing PR. CPU dispatch, storage, shape, stride, epsilon, fallback, and caller tests: **64 passed**; **90 CUDA tests** passed in the archived GPU run, including BF16/FP32, non-contiguous tensors, fullgraph, changing Graph inputs, and dynamic image/text lengths. CPU wrappers use real PyTorch with optional CUDA imports/launches replaced and are not GPU numerical evidence.

The fresh publication checkout passes all applicable pre-commit hooks for the six changed files, including mypy, CI markers, SPDX, and import checks. Full-repository pre-commit was also run; unrelated existing markdown, mypy, marker, shellcheck, and SPDX failures remain. No unrelated autoformat changes are included.

## Complete Qwen-Image requests

Hardware: one RTX PRO 5000 Blackwell 48 GB (SM120). Fixed Qwen-Image revision `75e0b4be04f60ec59a75f475837eced720f823b6`, TP1, BF16, TORCH_SDPA, component CPU offload, 1024×1024, 50 denoising steps, one fixed prompt and three seeds, prompt cache disabled.

Baseline is the exact earlier public PR head **45b4751**, loaded independently; these results are incremental over that implementation, not a claim against current main. Timing includes text encoding, every DiT step, VAE, and output readiness. The original native LayerNorm remains in both paths.

| Cohort | Baseline median request | Paired median request | Mean paired time reduction | Descriptive 95% range | Faster groups |
| --- | ---: | ---: | ---: | ---: | ---: |
| Three-version interleaved | 68.220 s | 66.937 s | 1.922% | 1.779–2.059% | 6/6 |
| Independent-process BA/AB confirmation | 65.260 s | 63.883 s | 2.095% | 1.931–2.250% | 8/8 |

The first cohort used six complete warmups and all six orders of baseline/intermediate/paired variants, followed by 18 timed requests. The confirmation used four complete warmups and eight BA/AB pairs, followed by 16 timed requests. All **44 complete generations** matched the independent public baseline element-for-element. All timed samples are retained; cohorts and uncertainty are reported separately. Percentages average paired observations and are not ratios of the displayed medians.

DiT paired time reductions were 2.141% and 2.260%. Peak allocated memory was **45,600,344,576 bytes** for both implementations, a **0-byte increase**. Confirmation peak reserved memory was **47,039,119,360 bytes** for both. Earlier sequential measurements were confounded by text-encoding drift and are not used for the performance claim. The independent confirmation compares both execution orders.

These measurements support this explicit configuration only; there is no multi-GPU, concurrent-throughput, other-model, or default-enablement claim. No new GPU run was performed during publication.

For CUDA regression reproduction in a configured project environment:

```bash
pytest tests/diffusion/layers/test_gated_residual_adaln.py \
  tests/diffusion/layers/test_gated_residual_adaln_v2.py \
  tests/diffusion/layers/test_paired_adaln.py -v
```

### CODE REVIEW

**Review by Dong1017 — COMMENTED** [review]
I reran `45b4751cb7af` on L20X (vLLM 0.28, PyTorch 2.13/CUDA 13, Triton 3.7.1): 66 tests passed, covering both outputs, rounding around native FP32 LayerNorm, strided inputs, fallback, compile and CUDA Graph replay. I didn't find a correctness issue in these checks.

The PR reports 28 current-head CPU tests, with 39 GPU tests and three Qwen-Image seed pairs at the earlier `158c3a992d6b`. My GPU run bypasses the A100 performance gate through the existing test fixture; it doesn't reproduce the A100 timings or full-model results. Whole-model compilation also remains untested here.

This matches the residual+AdaLN follow-up in [#7382](https://github.com/vllm-project/vllm-omni/issues/7382#issuecomment-5629791477), and its two-output contract is different from the basic gated-residual work in #6387. For the refactor, is the intended home a shared op under `layers.ops`, or a Qwen-local helper? It would be useful to settle that ownership without expanding this PR into the R1 migration.

File: .buildkite/cuda/test-ready.yml [hunk 1]
```diff
@@ -80,6 +80,7 @@ steps:
       - label: "Diffusion · Other Test"
         commands:
           - timeout 15m pytest -sv tests/diffusion/*.py tests/diffusion/ar_diffusion -m 'core_model and cuda' --run-level "core_model"
+          - timeout 10m pytest -sv tests/diffusion/layers/test_gated_residual_adaln.py -m 'core_model and cuda' --run-level "core_model"
```
Comment (lishunyang12) [inline .buildkite/cuda/test-ready.yml]: Run test_gated_residual_adaln_v2.py and test_paired_adaln.py in this CUDA job too.

Comment (0z5a) [reply .buildkite/cuda/test-ready.yml]: @lishunyang12 Fixed in 06cb90bb09915d9c5586e4ebd596f03c5d61a9ab — the CUDA job now runs test_gated_residual_adaln.py, test_gated_residual_adaln_v2.py, and test_paired_adaln.py together. I also reran the exact three-file CUDA command on an NVIDIA RTX PRO 4000 Blackwell (SM120): 90 passed, 64 deselected, 15 warnings in 17.46s (38 + 42 + 10 tests). Thanks!

**Comment by 0z5a** [comment]
Clarified in 4f8ac139: these remain shared ops in `vllm_omni.diffusion.layers`, with Qwen-Image as the current caller. Moving them into `layers.ops` can stay with the broader refactor. Reran all three AdaLN test files on L20: 154 passed. This validates the helper paths, not new A100 performance or whole-model compilation claims.
